doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -441,80 +425,140 @@ automatically created.
       ``!`` or ``-`` items that specify if an argument of the identifier is to be
       treated as an input (``+``), if its head only is an input (``!``) or an output
       (``-``) of the identifier. For a mode to match a list of arguments, input
       terms and input heads *must not* contain existential variables or be
-      existential variables respectively, while outputs can be any term. Multiple
-      modes can be declared for a single identifier, in that case only one mode
-      needs to match the arguments for the hints to be applied. The head of a term
+      existential variables respectively, while outputs can be any term.
+
+      The head of a term
       is understood here as the applicative head, or the match or projection
       scrutinee’s head, recursively, casts being ignored. :cmd:`Hint Mode` is
       especially useful for typeclasses, when one does not want to support default
       instances and avoid ambiguity in general. Setting a parameter of a class as an
-      input forces proof-search to be driven by that index of the class, with ``!``
-      giving more flexibility by allowing existentials to still appear deeper in the
-      index but not at its head.
+      input forces proof search to be driven by that index of the class, with ``!``
+      allowing existentials to appear in the index but not at its head.
 
    .. note::
 
-      + One can use a :cmd:`Hint Extern` with no pattern to do
-        pattern matching on hypotheses using ``match goal with``
-        inside the tactic.
+      + Multiple modes can be declared for a single identifier.  In that
+        case only one mode needs to match the arguments for the hints to be applied.
 
       + If you want to add hints such as :cmd:`Hint Transparent`,
         :cmd:`Hint Cut`, or :cmd:`Hint Mode`, for typeclass
         resolution, do not forget to put them in the
         ``typeclass_instances`` hint database.
 
+.. _hintrewrite:
+
+.. cmd:: Hint Rewrite {? {| -> | <- } } {+ @one_term } {? using @ltac_expr } {? : {* @ident } }
+
+   :n:`{? using @ltac_expr }`
+     If specified, :n:`ltac_expr` is applied to the generated subgoals, except for the
+     main subgoal.
+
+   :n:`{| -> | <- }`
+     Arrows specify the orientation; left to right (:n:`->`) or right to left (:n:`<-`).
+     If no arrow is given, the default orientation is left to right (:n:`->`).
+
+   Adds the terms :n:`{+ @one_term }` (their types must be
+   equalities) to the rewriting bases :n:`{*  @ident }`.
+   Note that the rewriting bases are distinct from the :tacn:`auto`
+   hint bases and that :tacn:`auto` does not take them into account.
+
+   This command is synchronous with the section mechanism (see :ref:`section-mechanism`):
+   when closing a section, all aliases created by :cmd:`Hint Rewrite` in that
+   section are lost. Conversely, when loading a module, all :cmd:`Hint Rewrite`
+   declarations at the global level of that module are loaded.
```

Comment:
This paragraph is not useful anymore now that this is presented in the same section as the other `Hint` command.